bytecomp/emitcode.ml

```diff
@@ -154,10 +154,21 @@ let slot_for_literal sc =
   enter (Reloc_literal (Symtable.transl_const sc));
   out_int 0
 and slot_for_getglobal id =
-  enter (Reloc_getglobal id);
+  let name = Ident.name id in
+  let reloc_info =
+    if Ident.is_predef id then (Reloc_getpredef (Predef_exn name))
+    else if Ident.global id then (Reloc_getcompunit (Compunit name))
+    else assert false
+  in
+  enter reloc_info;
   out_int 0
 and slot_for_setglobal id =
-  enter (Reloc_setglobal id);
+  let name = Ident.name id in
+  let reloc_info =
+    if Ident.global id then (Reloc_setcompunit (Compunit name))
```

Comment:
You want `Ident.persistent` rather than `Ident.global` here since `Ident.global` is true for the predef case too.